bluetooth: mesh: Use bt_get_name to get device name

PavelVPV opened the pull request:
By default if CONFIG_BT_MESH_PROXY_USE_DEVICE_NAME or CONFIG_BT_MESH_PB_GATT_USE_DEVICE_NAME is enabled, the mesh stack will add BT_DATA_NAME_COMPLETE AD Type along with the Mesh Proxy Service or Mesh Provisioning Service advertisements accordingly.

When BT_LE_ADV_OPT_USE_NAME was present and
CONFIG_BT_DEVICE_NAME_DYNAMIC is enabled, the advertised name was automatically updated by the host. This turned out to be a side-effect rather than expected behavior and after #71700 this behavior waa changed.

But customers use dynamic name feature.

This commit makes the mesh stack use bt_get_name to get the device name, which returns runtime name if CONFIG_BT_DEVICE_NAME_DYNAMIC is enabled.

PavelVPV committed:
bluetooth: mesh: Use bt_get_name to get device name

By default if CONFIG_BT_MESH_PROXY_USE_DEVICE_NAME
or CONFIG_BT_MESH_PB_GATT_USE_DEVICE_NAME is enabled, the mesh stack
will add BT_DATA_NAME_COMPLETE AD Type along with the Mesh Proxy Service
or Mesh Provisioning Service advertisements accordingly.

When BT_LE_ADV_OPT_USE_NAME was present and
CONFIG_BT_DEVICE_NAME_DYNAMIC is enabled, the advertised name was
automatically updated by the host. This turned out to be a side-effect
rather than expected behavior and after #71700 this behavior waa
changed.

But customers use dynamic name feature.

This commit makes the mesh stack use bt_get_name to get the device name,
which returns runtime name if CONFIG_BT_DEVICE_NAME_DYNAMIC is enabled.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

jhedberg commented on subsys/bluetooth/mesh/pb_gatt_srv.c:
```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_pb_gatt_srv);
 
 #define FAST_ADV_TIME (60LL * MSEC_PER_SEC)
 
+#if defined(CONFIG_BT_DEVICE_NAME_DYNAMIC)
+#define BT_DEVICE_NAME_LEN strlen(bt_get_name())
+#else
+#define BT_DEVICE_NAME_LEN (sizeof(CONFIG_BT_DEVICE_NAME) - 1)
+#endif
```
Did you consider putting variants of a `#define BT_DEVICE_NAME` into the conditional branches too, so that in the static case you don't end up resolving to a function call but just a string literal? I suspect the compiler will likely optimize this anyway since it should know that the function just returns the string literal, but since you have these conditional branches anyway you could remove some doubt by just having that extra define.

alxelax commented on subsys/bluetooth/mesh/pb_gatt_srv.c:
```diff
@@ -38,6 +38,12 @@ LOG_MODULE_REGISTER(bt_mesh_pb_gatt_srv);
 
 #define FAST_ADV_TIME (60LL * MSEC_PER_SEC)
 
+#define BT_DEVICE_NAME (IS_ENABLED(CONFIG_BT_DEVICE_NAME_DYNAMIC) ? \
```
I'm not big fun of copy-paste approach but didn't find correct header for this definition :(

PavelVPV replied:
Created a separate header file. Also moved macros for adv intervals as they are used by both pb_gatt_srv.c and proxy_srv.c.

PavelVPV committed:
bluetooth: mesh: gatt: Move generic GATT related defines to gatt.h

This moves GATT releated defines that are used by both Mesh Proxy
Service and Mesh Provisioning Service implementations to a common
header file.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

VincentFontanellaLinkio commented:
Can you push it on 3.7.0 that should be LTS and I consider as a major issue for us.
